sycl/test/regression/unnamed-lambda.cpp

```diff
@@ -0,0 +1,10 @@
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-device-only -c %s -o %t.temp
```

Comment:
@erichkeane, this test should probably be copied into #3894 to ensure that we have fixed the original problem

Reply:
Done, should be in the latest patch.